recipe/meta.yaml

```diff
@@ -0,0 +1,248 @@
+
+    test:
+      commands:
+        # headers
+        - test -f $PREFIX/include/arrow/api.h
+        - test -f $PREFIX/include/arrow/flight/types.h
+        - test -f $PREFIX/include/plasma/client.h
+        # - test -f $PREFIX/include/gandiva/engine.h 
+        - test -f $PREFIX/include/parquet/api/reader.h
+
+        # shared
+        - test -f $PREFIX/lib/libarrow${SHLIB_EXT}
+        - test -f $PREFIX/lib/libarrow_dataset${SHLIB_EXT}
+        - test -f $PREFIX/lib/libarrow_flight${SHLIB_EXT}
+        - test -f $PREFIX/lib/libarrow_python${SHLIB_EXT}
+        - test -f $PREFIX/lib/libparquet${SHLIB_EXT}
+        - test -f $PREFIX/lib/libplasma${SHLIB_EXT}
+        - test -f $PREFIX/lib/libarrow_cuda${SHLIB_EXT}    # [build_type == 'cuda']
+        - test ! -f $PREFIX/lib/libarrow_cuda${SHLIB_EXT}  # [build_type != 'cuda']   
+
+        # absence of static libraries
+        - test ! -f $PREFIX/lib/libarrow.a
+        - test ! -f $PREFIX/lib/libarrow_dataset.a
+        - test ! -f $PREFIX/lib/libarrow_flight.a
+        - test ! -f $PREFIX/lib/libarrow_python.a
+        - test ! -f $PREFIX/lib/libplasma.a
+        - test ! -f $PREFIX/lib/libparquet.a
+        # - test ! -f $PREFIX/lib/libgandiva.a
+
+        # conda tools
+        - conda inspect linkages -p $PREFIX $PKG_NAME
+
+  - name: pyarrow
+    script: build-pyarrow.sh
+    version: {{ version }}
+    build:
+      string: py{{ CONDA_PY }}h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}_{{ build_ext }}
+{% if build_type == 'cuda' %}
+      script_env:
+        - CUDA_HOME
+{% endif %}
+      ignore_run_exports:
+        - cudatoolkit
+      track_features:
+        {{ "- arrow-cuda" if build_type == 'cuda' else "" }}
+    requirements:
+      build:
+        - cmake
+        - ninja
+        - make
+        - {{ compiler('c') }}
+        - {{ compiler('cxx') }}
+      host:
+        - {{ pin_subpackage('arrow-cpp', exact=True) }}
+        - cython
+        - numpy {{ numpy }}
+        - python {{ python }}
+        - setuptools {{ setuptools }}
+        - setuptools_scm 5.0.2
+        - setuptools-scm 5.0.2
+        - six {{ six }}
+      run:
+        - {{ pin_subpackage('arrow-cpp', exact=True) }}
+        - {{ pin_compatible('numpy', lower_bound='1.16') }}
+        - python {{ python }}
+      run_constrained:
+        - arrow-cpp-proc * {{ build_ext }}
+        - cudatoolkit >=10.2  # [build_type == 'cuda']
+
+    about:
+      home: http://github.com/apache/arrow
+      license: Apache-2.0
+      license_file:
+        - LICENSE.txt
+      summary: Python libraries for Apache Arrow
+
+    test:
+      imports:
+        - pyarrow
+        - pyarrow.dataset
+        - pyarrow.flight
+        - pyarrow.orc
+        - pyarrow.parquet
+        - pyarrow.plasma
+        - pyarrow._hdfs
+        - pyarrow.cuda   # [build_type == 'cuda']
+      commands:
+        - test ! -f ${SP_DIR}/pyarrow/tests/test_array.py
+
+  - name: pyarrow-tests
+    script: build-pyarrow.sh
```

Comment:
Does the pyarrow-tests package also include the rest of pyarrow? If so, what is the purpose of creating two different packages?

Reply:
For build-time testing mostly. The pyarrow-tests package can run the `${SP_DIR}/pyarrow/tests/test_array.py` test, but we'd rather not include that in the "real" pyarrow package.

Reply:
Hmm OK. This will add an extra package to our channel that we probably don't want, and will end up in our generated environment files. We don't currently have a way to specify only wanting some of the packages from within a feedstock in our environment files. It would probably be best to create open-ce tests for pyarrow instead of this method. Can you please create an issue for that. We can merge in as is for now.